modules/public/api.py

```diff
@@ -99,17 +106,27 @@ def get_snap_details(snap_name):
         headers=DETAILS_QUERY_HEADERS
     )
 
-    if details_response.status_code >= 400:
-        message = (
-            'Failed to get snap details for {snap_name}'.format(**locals())
-        )
-
-        if details_response.status_code == 404:
-            message = 'Snap not found: {snap_name}'.format(**locals())
-
-        flask.abort(details_response.status_code, message)
-
-    return details_response.json()
+    try:
+        details = details_response.json()
+    except ValueError as decode_error:
+        error_message = ''.join([
+            "JSON decoding failed: ",
+            str(decode_error),
+        ])
+        raise InvalidResponseContent(error_message)
+
+    if details_response.status_code != 200:
+        if 'error_list' in details:
+            api_error_exception = ApiErrorResponse("Error list")
```

Comment:
You might get some expiration from https://git.launchpad.net/snapdevicegw/tree/snapdevicegw/exceptions.py and have a chain of exceptions that would treat the API errors internally, `APIError(error_response)`